### ISSUE TITLE
How can i solve this error: "/libexpat.so: undefined reference to `__isnanf'"

### ISSUE THREAD

**Issue by DZ2048** [libexpat/libexpat#511]
when i cross compile expat 2.4.1, show the error as following:
~~~~~~~~~~~~~~~~~~~~~~~~~
/bin/sh ../libtool  --tag=CC   --mode=link arm-hisiv100nptl-linux-gcc -std=gnu99 -Wall -Wextra -fexceptions -fno-strict-aliasing -Wmissing-prototypes -Wstrict-prototypes -pedantic -Wlogical-op -Wshadow -Wformat=2 -Wno-pedantic-ms-format -fvisibility=hidden -g -O2 -fno-strict-aliasing  -o elements elements.o ../lib/libexpat.la 
libtool: link: arm-hisiv100nptl-linux-gcc -std=gnu99 -Wall -Wextra -fexceptions -fno-strict-aliasing -Wmissing-prototypes -Wstrict-prototypes -pedantic -Wlogical-op -Wshadow -Wformat=2 -Wno-pedantic-ms-format -fvisibility=hidden -g -O2 -fno-strict-aliasing -o .libs/elements elements.o  ../lib/.libs/libexpat.so -Wl,-rpath -Wl,/home/expat-2.4.1/_install/lib
../lib/.libs/libexpat.so: undefined reference to `__isnanf'
collect2: ld returned 1 exit status
make[2]: *** [Makefile:412：elements] 错误 1
make[2]: 离开目录“/home/expat-2.4.1/examples”
make[1]: *** [Makefile:637：all-recursive] 错误 1
make[1]: 离开目录“/home/expat-2.4.1”
make: *** [Makefile:496：all] 错误 2
~~~~~~~~~~~~~~~~~~~~~~

**Comment by hartwork** [comment]
Hi @DZ2048 , there has been recent work on exactly this topic (see closed issue #509 and merged pull request #510). Could you try with Git `master`? The description of pull request #510 has instructions how to do that, just drop the `--branch issue-509-link-against-libm` part. If you get a chance to try both the CMake and the GNU Autotools ways (as described there) and report back if they work with your case, that would be appreciated. Looking forward to your reply.

**Comment by DZ2048** [comment]
> 
> 
> Hi @DZ2048 , there has been recent work on this exact topic (see closed issue #509 and merged pull request #510). Could you try with Git `master`? The description of pull request #510 has instructions how to do that, just drop the `--branch issue-509-link-against-libm` part. If you get a chance to try both the CMake and the GNU Autotools ways (as described there) and report back if they work with your case, that would be appreciated. Looking forward to your reply.

Thanku，but I can not find 'issue-510-link-against-libm' when i use git clone command.

**Comment by hartwork** [comment]
@DZ2048 correct, that branch was deleted after merging. Just drop the `--branch issue-509-link-against-libm` bits, i.e.

To test with GNU Autotools, locally:

```shell
cd "$(mktemp -d)"
git clone --depth 1 https://github.com/libexpat/libexpat
cd libexpat/expat/
./buildconf.sh && ./configure && make all check
```

To test with CMake, locally:

```shell
cd "$(mktemp -d)"
git clone --depth 1 https://github.com/libexpat/libexpat
cd libexpat/expat/
cmake . && make VERBOSE=1 all test
```

The `--depth 1` above is optional and you'll have to add your cross-compile bits of course. Let me know how things go.

**Comment by DZ2048** [comment]
> 
> 
> @DZ2048 correct, that branch was deleted after merging. Just drop the `--branch issue-509-link-against-libm` bits, i.e.
> 
> To test with GNU Autotools, locally:
> 
> ```shell
> cd "$(mktemp -d)"
> git clone --depth 1 https://github.com/libexpat/libexpat
> cd libexpat/expat/
> ./buildconf.sh && ./configure && make all check
> ```
> 
> To test with CMake, locally:
> 
> ```shell
> cd "$(mktemp -d)"
> git clone --depth 1 https://github.com/libexpat/libexpat
> cd libexpat/expat/
> cmake . && make VERBOSE=1 all test
> ```
> 
> The `--depth 1` above is optional and you'll have to add your cross-compile bits of course. Let me know how things go.

GOT it. please check following:
[](ur
[GNU-Autotools-check20210924.txt](https://github.com/libexpat/libexpat/files/7219289/GNU-Autotools-check20210924.txt)
l
[CMakeOutput.log](https://github.com/libexpat/libexpat/files/7219290/CMakeOutput.log)
[CMakeError.log](https://github.com/libexpat/libexpat/files/7219291/CMakeError.log)
)

**Comment by hartwork** [comment]
@DZ2048 thanks! So both worked fine on Git master? The way I read the CMake output there, some things cannot be auto-detected but that seems okay. What do you think?

**Comment by hartwork** [comment]
@DZ2048 PS: I notice now that the CMake logs seem to be about target `i686-linux-gnu` rather than `arm-hisiv100nptl-linux-gcc`?

**Comment by DZ2048** [comment]
oh,ye, i will use the arm-hisiv100nptl to run the test again.

**Comment by hartwork** [comment]
@DZ2048 any news?
